site/docs/evolution.md

```diff
@@ -54,6 +54,11 @@ Iceberg uses unique IDs to track each column in a table. When you add a column,
 
 Iceberg table partitioning can be updated in an existing table because queries do not reference partition values directly.
 
+When you evolve a partition spec, the old data written with an earlier spec remains unchanged. New data is written using the new spec in a new layout. Metadata for each of the partition versions is kept separately. Because of this, when you start writing queries, you get split planning. This is where each partition layout plans files separately using the filter it derives for that specific partition layout. Here's a visual representation of a contrived example: 
+
+![Partition evolution diagram](img/partition-evolution-diagram.png)
```

Comment:
I'm not sure that this diagram shows partition evolution. I agree with the description above, but the diagram appears to demonstrate pushdown. It is also slightly misleading because it uses "partition key" instead of "partition expression". Typically, partition key is a tuple of data that identifies a specific partition. Also, "between" is not really supported because it is ambiguous so it would be better to use inequalities and AND.

Reply:
I see what you mean regarding the first point. This diagram was drawn up when we were trying to understand how exactly partition evolution would work with queries (I think we emailed you a while back and you explained the split planning approach to us which led to this diagram). Would it be worth refining this diagram a little so it can be a visual example of the split planning concept? I'll have a think about another diagram that might fit better here too :)

Reply:
Yeah, I think it's possible to update the diagram. Maybe having two separate copies that each has a different granularity for time?